storage/entity.go

```diff
@@ -0,0 +1,350 @@
+package storage
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"io/ioutil"
+
+	"github.com/satori/uuid"
+)
+
+// Annotating as secure for gas scanning
+/* #nosec */
+const (
+	partitionKeyNode  = "PartitionKey"
+	rowKeyNode        = "RowKey"
+	etagErrorTemplate = "Etag didn't match: %v"
+)
+
+// Entity represents an entity inside an Azure table.
+type Entity struct {
+	tsc           *TableServiceClient
```

Comment:
If the Table field is never allowed to be nil you could use the TableServiceClient from that field (this is just off the cuff, I haven't completely digested the APIs so perhaps this doesn't make sense).